### PR TITLE
Fixes #16389 - enable taxable object creation in specific context

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

File: app/models/concerns/taxonomix.rb [hunk 1]
```diff
@@ -36,7 +36,8 @@ def with_taxonomy_scope(loc = Location.current, org = Organization.current, inne
       self.which_location        = Location.expand(loc) if SETTINGS[:locations_enabled]
       self.which_organization    = Organization.expand(org) if SETTINGS[:organizations_enabled]
       scope = block_given? ? yield : where('1=1')
-      scope = scope.where(:id => taxable_ids) if taxable_ids
+      tax_ids = taxable_ids
+      scope = scope.where(:id => tax_ids) if tax_ids
```
Comment (ares) [inline app/models/concerns/taxonomix.rb]: unrelated performance boost, taxable_ids is not cached

**Comment by ehelms** [comment]
[test]

**Comment by iNecas** [comment]
APJ, if there was emoji for :respect: I would use one :)

**Comment by ares** [comment]
@iNecas one more change was needed, we have to be explicit about the table name and we can't escape the values inside `IN ()`, these injections should be always safe though. Please take another look, thanks.

**Comment by dLobatog** [comment]
Merged as 35ed04f95f14c934ffebc874e90a3a4a6684fe71, thanks @ares!

**Pull request closed without merging**